Formula/docker-credential-magic.rb

```diff
@@ -0,0 +1,63 @@
+class DockerCredentialMagic < Formula
+  desc "Magic shim for Docker credential helpers"
+  homepage "https://github.com/docker-credential-magic/docker-credential-magic"
+  license any_of: ["Apache-2.0", "CC-BY-4.0"]
+
+  depends_on "go" => :optional
+
+  on_macos do
+    if Hardware::CPU.arm?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Darwin_arm64.tar.gz"
+      sha256 "47085f5ccc5dbd0c77e9ec5152995a901d4efc5d148a9e38df28500f83a15eba"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+    if Hardware::CPU.intel?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Darwin_x86_64.tar.gz"
+      sha256 "b50ec9c7b51afdd20424f8a31dbb0992b40bc597ef23ba7df8c6c3e81b86a027"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+  end
+
+  on_linux do
+    if Hardware::CPU.arm? && !Hardware::CPU.is_64_bit?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_armv6.tar.gz"
+      sha256 "fa6deaf643623a163fafdc5e27717207fc2c9fc13da03e59a001194c9f951f41"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+    if Hardware::CPU.arm? && Hardware::CPU.is_64_bit?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_arm64.tar.gz"
+      sha256 "4ec0f0630c1d91a9025f71f34dbbf678bc2553fcae956d6927d0431d70e0bf68"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+    if Hardware::CPU.intel?
+      url "https://github.com/docker-credential-magic/docker-credential-magic/releases/download/v0.8.0/docker-credential-magic_Linux_x86_64.tar.gz"
+      sha256 "afbb0aacea8b5cbdf9f99cd3ec5a9f649ee421866ec50894bc83ea30e0b812f9"
+
+      def install
+        bin.install "docker-credential-magic"
+        bin.install "docker-credential-magician"
+      end
+    end
+  end
+
+  test do
+    system "#{bin}/docker-credential-magic", "version"
+    system "#{bin}/docker-credential-magician", "version"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
got it

Reply:
This still needs to be done.